### ISSUE TITLE
[turborepo] Crash in Windows

### ISSUE THREAD

**Issue by unional** [vercel/turbo#2853]
### What version of Turborepo are you using?

1.6.3

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Windows

### Describe the Bug

It suddenly crashes on one of the repo (still working on other repos):

```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x14bf14c]

goroutine 8 [running]:
github.com/vercel/turbo/cli/internal/daemon/connector.(*Connector).getOrStartDaemon(0xc0015b1b58)
        github.com/vercel/turbo/cli/internal/daemon/connector/connector.go:287 +0xcc
github.com/vercel/turbo/cli/internal/daemon/connector.(*Connector).connectInternal(0x0?, {0x1934458, 0xc00003a0c8})
        github.com/vercel/turbo/cli/internal/daemon/connector/connector.go:219 +0x74
github.com/vercel/turbo/cli/internal/daemon/connector.(*Connector).Connect(0xc0015b1b58, {0x1934458?, 0xc00003a0c8?})
        github.com/vercel/turbo/cli/internal/daemon/connector/connector.go:200 +0x2c
github.com/vercel/turbo/cli/internal/daemon.GetClient({0x1934458, 0xc00003a0c8}, {0xc000035c40, 0x19}, {0x19385b8, 0xc0001a8ab0}, {0x17f7ded, 0x5}, {0x0, 0x0})
        github.com/vercel/turbo/cli/internal/daemon/daemon.go:304 +0x1da
github.com/vercel/turbo/cli/internal/run.(*run).run(0xc000005680, {0x1934458, 0xc00003a0c8}, {0xc0000ab170, 0x1, 0x1})
        github.com/vercel/turbo/cli/internal/run/run.go:211 +0x350
github.com/vercel/turbo/cli/internal/run.GetCmd.func1(0xc000233180, {0xc0000ab170, 0x1, 0x1})
        github.com/vercel/turbo/cli/internal/run/run.go:119 +0x24d
github.com/spf13/cobra.(*Command).execute(0xc000233180, {0xc00008e060, 0x1, 0x2})
        github.com/spf13/cobra@v1.3.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00021b180)
        github.com/spf13/cobra@v1.3.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.3.0/command.go:902
github.com/vercel/turbo/cli/internal/cmd.RunWithArgs.func1()
        github.com/vercel/turbo/cli/internal/cmd/root.go:54 +0x2f
created by github.com/vercel/turbo/cli/internal/cmd.RunWithArgs
        github.com/vercel/turbo/cli/internal/cmd/root.go:53 +0x2b0
```

### Expected Behavior

Not crashing

### To Reproduce

This is the repo: https://github.com/justland/just-web

Do not know if you will see the same error. The issue persist for me even if I delete the repo and clone again.

Just do:

```sh
pnpm i
pnpm verify # or any other script using turbo
```

### Reproduction Repo

https://github.com/justland/just-web

**Comment by unional** [comment]
After restarting Windows, it starts working. Keeping this open to see if you can figure out what to guard based on the backtrace.

**Comment by adam-coster** [comment]
This also happened to me, and was also resolved by restarting my Windows PC.

I **unsuccessfully tried resolving** the issue with the following:

- ❌ Deleting Turbo's caches
- ❌ Re-installing Turbo via `yarn`
- ❌ Closing & re-opening VS Code
- ❌ Killing all Node.js and console host instances (via Task Manager)

✅ Only rebooting my PC resolved the issue.

I was **unable to reproduce** the issue. I tried the following:

- ❌ Closing & re-opening VS Code while running Turbo
- ❌ Killing `turbo.exe` via Task Manager
- ❌ Killing random Node.js and console host process spawned by Turbo, while it was running
- ❌ `Ctrl-C`- terminating Turbo via the command line

My context:

- Windows 11 Pro 22H2
- VSCode 1.74.0-insider
- Node.js v16.18.0
- yarn@3.2.4
- turbo@1.6.3 (running via `yarn turbo run <etc>`)

**Comment by teddybee** [comment]
Avira antivirus detect a malware in turbo.exe and puts it in quarantine.
https://www.virustotal.com/gui/file/30d3c052a097f3984b6eb64985a473b2538f1ea51955db950cf94b2354e02928

**Comment by caspercasanova** [comment]
Unsure what the problem is, I wonder if upgrading to windows 11 will fix the issue? The error message appeared all of a sudden after running `pnpm turbo build`

**Comment by unional** [comment]
> if upgrading to windows 11 will fix the issue?

I am on Windows 11

**Comment by subgan82** [comment]
We also saw this issue after upgrading to latest turborepo version on windows. It does not happen on Mac.

**Comment by ta834n** [comment]
Same here

Windows 10
NodeJS v18.12.1
turbo 1.6.3
pnpm 7.21.0

Restarting windows solved the issue for me.

**Comment by nathanhammond** [comment]
This was addressed by #2557.
